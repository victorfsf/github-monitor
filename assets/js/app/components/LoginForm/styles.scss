.login {
    padding-top: 10%;
    background-color: #fafafa;
    .login-box {
        padding: 1em;
        text-align: center;
    }
    .logo-container {
        padding-bottom: 15%;
        font-size: 2.5em;
    }
    .title {
        font-weight: 600;
    }
    .description {
        font-size: 0.5em;
    }
}
