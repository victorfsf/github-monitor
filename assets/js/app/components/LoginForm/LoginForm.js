import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LoginForm = props => (
  <div className="login">
    <div className="col-lg-4 col-md-6 col-sm-8 col-xs-12 col-centered">
      <div className="login-box">
        <div className="logo-container">
          <i className="fa fa-4x fa-github" />
          <div className="title">
            Github Monitor
          </div>
          <div className="description">
            {"Monitor your repositories' commits"}
          </div>
        </div>
        <a href={props.loginUrl} className="btn btn-dark">
          <i className="fa fa-2x fa-github-alt align-middle" />
          <span className="align-middle pl-2">
            Sign in with Github
          </span>
        </a>
      </div>
    </div>
  </div>
);

LoginForm.propTypes = {
  loginUrl: PropTypes.string.isRequired,
};

export default LoginForm;
