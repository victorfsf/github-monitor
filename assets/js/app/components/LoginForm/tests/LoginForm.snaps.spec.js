import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from 'app/components/LoginForm';


describe('LoginForm', () => {
  let Component;
  let tree;

  test('Form renders without url', () => {
    global.console = {
      error: jest.fn(),
    };
    Component = renderer.create(<LoginForm />);
    tree = Component.toJSON();
    expect(global.console.error).toHaveBeenCalledWith(
      'Warning: Failed prop type: The prop `loginUrl` is marked as required ' +
      'in `LoginForm`, but its value is `undefined`.\n    in LoginForm',
    );
  });

  test('Form renders with url', () => {
    Component = renderer.create(<LoginForm loginUrl="/login/url/" />);
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
