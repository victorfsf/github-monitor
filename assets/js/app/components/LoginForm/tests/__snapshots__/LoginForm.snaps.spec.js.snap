// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Form renders with url 1`] = `
<div
  className="login"
>
  <div
    className="col-lg-4 col-md-6 col-sm-8 col-xs-12 col-centered"
  >
    <div
      className="login-box"
    >
      <div
        className="logo-container"
      >
        <i
          className="fa fa-4x fa-github"
        />
        <div
          className="title"
        >
          Github Monitor
        </div>
        <div
          className="description"
        >
          Monitor your repositories' commits
        </div>
      </div>
      <a
        className="btn btn-dark"
        href="/login/url/"
      >
        <i
          className="fa fa-2x fa-github-alt align-middle"
        />
        <span
          className="align-middle pl-2"
        >
          Sign in with Github
        </span>
      </a>
    </div>
  </div>
</div>
`;
